### PR TITLE
plots: allow top-level strings

### PR DESCRIPTION
**Pull request by dberenbaum**
Follow up to https://github.com/iterative/dvc/pull/8412 to allow for entering file paths as strings instead of dict keys:

```diff
plots:
+  - path/to/file
-  - path/to/file:
```

Some of the JSON output tests are failing in `tests/integration/plots/test_plots.py`. @pared When you have a chance, could you take a look and see if you have any idea why?

### CODE REVIEW

**Comment by dberenbaum** [comment]
Schema changes are in https://github.com/iterative/dvcyaml-schema/tree/top-level-plot-str

**Comment by dberenbaum** [comment]
Docs PR: https://github.com/iterative/dvc.org/pull/4081

File: dvc/repo/plots/__init__.py [hunk 1]
```diff
@@ -444,16 +445,23 @@ def _collect_pipeline_files(repo, targets: List[str], props, onerror=None):
...
-                }
+                for elem in dvcfile_defs:
+                    if isinstance(elem, str):
+                        dvcfile_defs_dict[elem] = None
+                    else:
+                        k, v = list(elem.items())[0]
+                        dvcfile_defs_dict[k] = v
+            else:
+                dvcfile_defs_dict = dvcfile_defs
+            print(dvcfile_defs_dict)
```
Comment (pared) [inline dvc/repo/plots/__init__.py:458]: this is the cause :)

Comment (pared) [inline dvc/repo/plots/__init__.py:458]:
```suggestion

```

**Review by pared — CHANGES_REQUESTED** [review]

File: dvc/repo/plots/__init__.py (hunk 1 above)
Comment (dberenbaum) [reply dvc/repo/plots/__init__.py]: 😬 Sorry about that!

**Review by pared — APPROVED** [review]

**Comment by dberenbaum** [comment]
@pared Could you merge when you have a chance? I don't have permission.

**Comment by skshetry** [comment]
> @pared Could you merge when you have a chance? I don't have permission.

@dberenbaum, please check. I have fixed it (`all` had `write` permissions but somehow `dvc-all` had lesser permissions which I have switched).

**Comment by dberenbaum** [comment]
Thanks @skshetry. TBH I don't mind not having merge permissions (it might be a good separation of duties) but I think `dvc-all` would need `maintain` privileges to merge to main. Looks like I'm still blocked right now from merging.

**Comment by skshetry** [comment]
@dberenbaum, `Write` does have merge permissions. See https://docs.github.com/en/organizations/managing-user-access-to-your-organizations-repositories/repository-roles-for-an-organization#permissions-for-each-role.

What you may not have is
>  Merge pull requests on protected branches, even if there are no approving reviews

which requires Admin permissions.

**Pull request merged**